Update CSES Movie Festival II.mdx (#780)
cpinitiative/usaco-guide: solutions/CSES Movie Festival II.mdx

id: cses-1632
source: CSES
title: Movie Festival II
author: Shreyas Thumathy, Benjamin Qi
---

<!-- In this problem, we're asked the maximum number of movie intervals $(start, end)$ we can cover using $k$ people. -->

<!-- ## Main Idea -->

The first step is the same as that of [Movie Festival](/silver/greedy-sorting#example---the-scheduling-problem); sort the movies in increasing order of end time. For each movie in order, we will assign it to one of the $K$ members to watch (or none of them).

> Keep track of the time each member finishes watching the latest movie that has been assigned to them in a sorted collection (e.g. a `multiset` in C++). Initially, the collection consists of $K$ zeroes.

Reviewer: ```suggestion
Keep track of the time each member finishes watching all of his currently assigned movies in a sorted collection (e.g. a [`multiset`](https://en.cppreference.com/w/cpp/container/multiset) in C++). Initially, the collection consists of $K$ zeroes.
```

> For each movie in order, we can assign a member to watch it if there exists an element in the sorted collection less than or equal to the start time of the movie. If there are multiple such elements, choose the greatest one (the member who finished watching the latest movie assigned to him the latest). Increment the answer and update the collection accordingly.

Reviewer: ```suggestion
For each movie in order, we can assign a member to watch it if there exists an element in the sorted collection less than or equal to the start time of the movie. If there are multiple such elements, choose the greatest one (the member who finished watching his assigned movies the latest). Assign the member to watch this movie by incrementing the answer and updating the collection accordingly.
```

<!-- If there are multiple members, choose the one who finishes watching the previous movie latest (that way, if we encounter a movie that has an earlier start time, we can still assign that movie). -->

<LanguageSection>

<CPPSection>

## C++ Implementation

```cpp
#include <iostream>
#include <vector>
#include <set>
#include <algorithm>
using namespace std;

int main() {
	int n, k; cin >> n >> k;

> 	vector<pair<int, int>> v(n);

Reviewer: Technically pair is in `#include<utility>`, though I don't know how cpp headers work. Including `iostream`, `vector`, and `set` all automatically allow use of `pair`s but `chrono` doesn't.
